ask-sdk-core/src/com/amazon/ask/dispatcher/impl/DefaultRequestDispatcher.java

```diff
@@ -131,17 +130,31 @@ protected DefaultRequestDispatcher(Collection<HandlerAdapter> handlerAdapters, C
             throw new AskSdkException(message);
         }
 
-        // execute any request interceptors attached to the handler chain
-        for (RequestInterceptor requestInterceptor : handlerChain.get().getRequestInterceptors()) {
-            requestInterceptor.process(input);
-        }
+        Optional<Response> response;
+        try {
+            // execute any request interceptors attached to the handler chain
+            for (RequestInterceptor requestInterceptor : handlerChain.get().getRequestInterceptors()) {
+                requestInterceptor.process(input);
+            }
 
-        // invoke request handler using the adapter
-        Optional<Response> response = handlerAdapter.execute(input, requestHandler);
+            // invoke request handler using the adapter
+            response = handlerAdapter.execute(input, requestHandler);
 
-        // first execute any response interceptors attached to the handler chain, then any global ones
-        Stream.concat(handlerChain.get().getResponseInterceptors().stream(), responseInterceptors.stream())
-                .forEach(responseInterceptor -> responseInterceptor.process(input, response));
+            // execute any response interceptors attached to the handler chain
+            for (ResponseInterceptor responseInterceptor : handlerChain.get().getResponseInterceptors()) {
+                responseInterceptor.process(input, response);
+            }
+        } catch (Exception e) {
+            return handlerChain.get().getExceptionHandlers().stream()
```

Comment:
What's the contract for a `RuntimeException` thrown by a `ResponseInterceptor`? I see three options:
1. Handle them with a `ExceptionHandler` .. *This is what we do right now.*
2. Swallow and log them
3. Fail the request.

I ask because it seems wrong to replace a valid `Response` from the `RequestHandler` with one from an `ExceptionMapper` just because of a rogue `ResponseInterceptor`.

Reply:
I think it's valid to surface these exceptions up to the ExceptionHandler since Skill devs can use response interceptors for response sanitization or attribute storage. That said I could also see cases where you wouldn't want to fail for non-critical stuff like logging. We can't change the default behavior at this point but I would not be opposed to making it configurable in the future.